packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -529,3 +529,32 @@ export function getHttpOperationResponseHeaders(
   }
   return headers;
 }
+
+export class DuplicateTracker<K, V> {
+  #entries = new Map<K, Set<V>>();
```

Comment:
Can't use the existing DuplicateTracker, because we need `Set<V>` as value instead of `V[]`.
Otherwise, we will throw errors on the same model multiple times.

Reply:
use the default one since we throw per scope.